### PR TITLE
Jonah Park javascript-timers Exercise

### PR DESCRIPTION
**Pull request by jonahp001**
Kapture 2023-02-22 at 09 43 01

### CODE REVIEW

File: javascript-timers/interval.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+function countDownEarth() {
+  if ($h1.textContent === '4') {
+    $h1.textContent = '3';
+  } else if ($h1.textContent === '3') {
+    $h1.textContent = '2';
+  } else if ($h1.textContent === '2') {
+    $h1.textContent = '1';
+  } else if ($h1.textContent === '1') {
+    $h1.textContent = '~Earth Beeeelooowww Us~';
+  }
```
Comment (robgmerrill) [inline javascript-timers/interval.js:12]: Within this final else if statement we are going to want to cancel setInterval from running. 

If you added a console log at the top of `countDownEarth` you would see that setInterval is still running after you have displayed your final text. You will need to clear that with `clearInterval`

```
function countDownEarth() { 
  console.log('i am still running');
  if ($h1.textContent === '4') {
    $h1.textContent = '3';
  } else if ($h1.textContent === '3') {
    $h1.textContent = '2';
  } else if ($h1.textContent === '2') {
    $h1.textContent = '1';
  } else if ($h1.textContent === '1') {
    $h1.textContent = '~Earth Beeeelooowww Us~';
    // call clearInterval() and pass in timer id. 
  }
  ```

File: javascript-timers/interval.js [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+setInterval(countDownEarth, 1000);
```
Comment (robgmerrill) [inline javascript-timers/interval.js:15]: You will need the timer id of the setInterval function. You can capture that with:

`var timerId = setInterval(countDownEarth, 1000);`

**Review by robgmerrill — CHANGES_REQUESTED** [review]
Your code will need the clearInterval method to stop setInterval from running.

File: javascript-timers/interval.js (hunk 1 above)
Comment (jonahp001) [reply javascript-timers/interval.js:12]: ah i see-- i put a console log at the end of it and i didnt see anything in the console, so i thought it was in the clear haha

File: javascript-timers/interval.js [hunk 3]
```diff
@@ -10,6 +10,8 @@ function countDownEarth() {
   } else if ($h1.textContent === '1') {
     $h1.textContent = '~Earth Beeeelooowww Us~';
   }
+  var countDownId = setInterval(countDownEarth, 1000);
```
Comment (robgmerrill) [inline javascript-timers/interval.js:13]: Close, but that variable should be set on line 17. On line 13 you unnecessarily are introducing a whole new setInterval function call inside of the else if. 

So remove this line: `var countDownId = setInterval(countDownEarth, 1000);`

**Review by robgmerrill — CHANGES_REQUESTED** [review]
Close. We aren't defining the timerId in the function countDown. It should be defined when when initially using setInterval at the bottom of the file.

**Comment by jonahp001** [comment]
Kapture 2023-02-22 at 10 13 29
i think i got it now haha

**Review by robgmerrill — APPROVED** [review]
👍

**Pull request merged**